src/pages/resultsView/pathwayMapper/ResultsViewPathwayMapper.tsx

```diff
@@ -0,0 +1,198 @@
+                                this.storeForAllData.samples.isComplete &&
+                                this.storeForAllData.patients.isComplete &&
+                                this.storeForAllData.coverageInformation.isComplete &&
+                                this.storeForAllData.sequencedSampleKeysByGene.isComplete &&
+                                this.storeForAllData.sequencedPatientKeysByGene.isComplete &&
+                                this.storeForAllData.selectedMolecularProfiles.isComplete;
+
+        if(isNewStoreReady){
+
+            this.storeForAllData!.oqlFilteredCaseAggregatedDataByUnflattenedOQLLine.result!.forEach( (alterationData, trackIndex) => {
+                getAlterationData(this.storeForAllData!.samples.result,
+                                  this.storeForAllData!.patients.result,
+                                  this.storeForAllData!.coverageInformation.result,
+                                  this.storeForAllData!.sequencedSampleKeysByGene.result!,
+                                  this.storeForAllData!.sequencedPatientKeysByGene.result!,
+                                  this.storeForAllData!.selectedMolecularProfiles.result!,
+                                  alterationData,
+                                  false,
+                                  this.alterationFrequencyData,
+                                  this.props.store.genes.result!);
+                
+            });
+
+            this.addGenomicData(this.alterationFrequencyData);
+            // Toasts are removed with delay
+            setTimeout(() => {this.activeToasts.forEach(tId => {toast.dismiss(tId);});}, 1000);
+        }
+
+        return(
+
+            <div className="pathwayMapper">
+            <div data-test="pathwayMapperTabDiv" className="cBioMode" style={{width: "99%"}}>
+                <Row>
+
+                    { !this.isLoading ? 
+
+                        <React.Fragment>
+                            <OqlStatusBanner className="coexp-oql-status-banner" store={this.props.store} tabReflectsOql={true}/>
+
+                            <PathwayMapper
+                                isCBioPortal={true}
+                                isCollaborative={false}
+                                genes={this.props.store.genes.result as any}
+                                cBioAlterationData={this.alterationFrequencyData}
+                                queryParameter={QueryParameter.GENE_LIST}
+                                oncoPrintTab={ResultsViewTab.ONCOPRINT}
+                                changePathwayHandler={this.changePathwayHandler}
+                                addGenomicDataHandler={this.addGenomicDataHandler}
+                                tableComponent={PathwayMapperTable}
+                                validGenes={this.validGenes}
+                                toast={toast}
```

Comment:
i understand toats are some kind of growl-like notification.  What are they used for here?

Reply:
Toast variable is sent to the PathwayMapper component so that when it needs to show notification it will use this one.